content/docs/command-reference/gc.md

```diff
@@ -44,8 +44,29 @@ If the `--cloud` (`-c`) flag is used, this command deletes unused data from the
 to deleting it from the local DVC cache. To specify a DVC remote to delete from,
 use the `--remote` (`-r`) option.
 
-> ⚠️ Danger: cloud deletion is irreversible unless there is another DVC remote
-> or a manual backup with the same data.
+<admon type="warn">
+
+Cloud deletion is irreversible unless there is another DVC remote or a manual
+backup with the same data.
+
+</admon>
+
+### Cleaning shared cache or shared remote
+
+If a single remote or a single [cache is shared] among different projects and
+those projects have overlapped DVC-tracked files, removing files via `gc` in one
+project may break some links in another project.
+
+To keep overlapping files `dvc gc` has a flag `--projects` (`-p`) that takes
+multiple path to DVC projects. DVC _keeps_ data referenced in those projects.
+Make sure that all the commits (including branches) that are referencing any
+files to be preserved have been pulled before running this command.
```

Comment:
```suggestion
To prevent this, use the `--projects` (`-p`) option. It takes one or more
paths to the DVC projects whose data should be preserved.
Make sure that all the commits and branches that reference
files you want to keep have been pulled in those other projects first.
```

Reply:
p.s. Wow this is a complicated process. I wonder whether there could be a better design e.g. where external caches"knew" what projects use them so you could simply supply `--project` to collect only the data used in the present project cc @dberenbaum

Reply:
Yup, something similar was mentioned by @iesahin about remote in https://github.com/iterative/dvc/discussions/6653. We could open a separate issue for that specific request. It's probably not something we are likely to prioritize soon since this is a command that's hopefully used sparingly, and there are a lot of complexities in solving it, but I agree it's not ideal as is.